utils/kubernetes/client.go

```diff
@@ -32,40 +33,71 @@ func DetectKubeConfig(configfile []byte) (config *rest.Config, err error) {
 	// Look for kubeconfig from the path mentioned in $KUBECONFIG
 	kubeconfig := os.Getenv("KUBECONFIG")
 	if kubeconfig != "" {
-		if config, err = clientcmd.BuildConfigFromFlags("", kubeconfig); err == nil {
+		_, cfgFile, err := ProcessConfig(kubeconfig, "")
+		if err != nil {
+			return nil, err
+		}
+		if config, err = clientcmd.RESTConfigFromKubeConfig(cfgFile); err == nil {
 			return config, err
 		}
 	}
 
 	// Look for kubeconfig at the default path
 	path := filepath.Join(utils.GetHome(), ".kube", "config")
-	if config, err = clientcmd.BuildConfigFromFlags("", path); err == nil {
+	_, cfgFile, err := ProcessConfig(path, "")
+	if err != nil{
+		return nil, err
+	}
+	if config, err = clientcmd.RESTConfigFromKubeConfig(cfgFile); err == nil {
 		return config, err
 	}
 
 	return
 }
 
-func processConfig(configFile []byte) ([]byte, error) {
-	cfg, err := clientcmd.Load(configFile)
+// ProcessConfig handles loading, validating, and optionally saving or returning a kubeconfig
+func ProcessConfig(kubeConfig interface{}, outputPath string) (*clientcmdapi.Config, []byte, error) {
+	var config *clientcmdapi.Config
+	var err error
+
+	// Load the Kubeconfig 
+	switch v := kubeConfig.(type) {
+	case string:
+		config, err = clientcmd.LoadFromFile(v)
+	case []byte:
+		config, err = clientcmd.Load(v)
+	default:
+		return nil, nil, ErrLoadConfig(err)
+	}
 	if err != nil {
-		return nil, ErrLoadConfig(err)
+		return nil, nil, err
```

Comment:
Please reuse/define a new meshkit error.

Reply:
Alright Noted! Thank you for the review 😃